src/net/java/sip/communicator/plugin/desktoputil/plaf/SearchFieldUI.java

```diff
@@ -177,10 +179,12 @@ protected void customPaintBackground(Graphics g)
                 - searchIcon.getIconHeight()/2;
 
             g2.drawImage(searchIcon.getImage(), 3, dy, null);
+            Pattern p = Pattern.compile(".*[a-zA-Z].*");
+            Matcher m = p.matcher(c.getText());
 
             if (c.getText() != null
```

Comment:
Use isNotEmpty(c.getText()) && !StringUtils.isAlpha(...)

Not sure what the test actually wants to check. Numbers only? Then the regex should be "\\d+". If it's meant to be a phone number, what about spaces, + for international, etc.

Reply:
By old code check wanted to know is there letters in the string or not

Reply:
May be [isAlphanumericSpace](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#isAlphanumericSpace-java.lang.CharSequence-)?